plugin/pkg/scheduler/algorithm/predicates/error.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import "fmt"
+
+var (
+	ErrExceededMaxPodNumber   = newResourceUnfitError("ExceededMaxPodNumber")
+	ErrInsufficientFreeCPU    = newResourceUnfitError("InsufficientFreeCPU")
+	ErrInsufficientFreeMemory = newResourceUnfitError("InsufficientFreeMemory")
```

Comment:
The below makes more sense to me.

```
 	ErrExceededMaxPodNumber   = newResourceUnfitError("PodCount")
 	ErrInsufficientFreeCPU    = newResourceUnfitError("CPU")
	ErrInsufficientFreeMemory = newResourceUnfitError("Memory")
```


I was trying to say, given your usage (only one parameter), the below will work for you, too:

```
 	ErrExceededMaxPodNumber   = errors.New("Insufficient resources: PodCount")
 	ErrInsufficientFreeCPU    = errors.New("Insufficient resources: CPU")
	ErrInsufficientFreeMemory = errors.New("Insufficient resources: Memory")
```

You *can* compare errors, i.e., `err == ErrExceededMaxPodNumber` if you do this, and it will work like you want.